applications/flash_attention_v2/kernel/xe_flash_attn_decode.hpp

```diff
@@ -296,6 +298,28 @@ class FMHADecode {
       auto pVgV = thr_prefetch_V.partition_S(gV_prefetch);
 
       int kv_tile_idx = sub_group_id / ATOM_N;
+      int kv_cache_tile_idx = kv_tile_idx;
+      int tiles_per_page = ceil_div(mainloop_params.page_size, QK_SG_N);
+
+      if constexpr (PagedKV) {
+        if (seq_len_kv_cache != 0) {
+          // get physical page idx from page table
+          int curr_batch_pages = is_var_len ? mainloop_params.num_pages_per_seq[batch_coord + 1] - mainloop_params.num_pages_per_seq[batch_coord]
+                                            : ceil_div(seq_len_kv_cache, mainloop_params.page_size);
+          int curr_page_logical_idx = (kv_tile_idx * QK_SG_N) / mainloop_params.page_size;
+          int batch_offset = is_var_len ? mainloop_params.num_pages_per_seq[batch_coord] : batch_coord * curr_batch_pages;
+          bool valid_page = curr_page_logical_idx < curr_batch_pages;
+          if (valid_page) {
+            kv_cache_tile_idx = mainloop_params.ptr_page_table[
+                  batch_offset +                     // page table for this batch
+                  curr_page_logical_idx              // tile idx to logical page idx
+                  ] * tiles_per_page +               // base block idx of physical page
+                  (kv_tile_idx % tiles_per_page);    // offset within page
```

Comment:
```suggestion
                  kv_tile_idx % tiles_per_page;    // offset within page
```
redundant parenthesis